File: VB/HowToSetMaskForCustomWinPropertyEdtior.Module/DomainObject1.vb
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

<DefaultClassOptions()> _
Public Class DomainObject1
    Inherits BaseObject
    Public Sub New(ByVal session As Session)
        MyBase.New(session)
    End Sub
    Private fNumber As Decimal

    Public Property Number() As Decimal
        Get
            Return fNumber
        End Get
        Set(ByVal value As Decimal)
            SetPropertyValue("Number", fNumber, Value)
        End Set
    End Property

End Class

File: CS/HowToSetMaskForCustomWinPropertyEdtior.Module/DomainObject1.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace HowToSetMaskForCustomWinPropertyEdtior.Module {
   [DefaultClassOptions]
   public class DomainObject1 : BaseObject {
      public DomainObject1(Session session) : base(session) { }
      private decimal number;

      public decimal Number {
         get { return number; }
         set { SetPropertyValue("Number", ref number, value); }
      }

   }

}
